chromadb/test/property/test_embeddings.py

```diff
@@ -90,8 +94,36 @@ def initialize(self, collection: strategies.Collection):
         self.record_set_state = strategies.StateMachineRecordSet(
             ids=[], metadatas=[], documents=[], embeddings=[]
         )
+        if self.__class__.__name__ == "EmbeddingStateMachine":
+            self.log_operation_count = 0
+            self.collection_version = self.collection.get_model()["version"]
+
+    @precondition(
+        lambda self: not NOT_CLUSTER_ONLY
+        and self.log_operation_count > 10
+        and self.__class__.__name__ == "EmbeddingStateMachine"
```

Comment:
What is the intention of this class name check? Not a huge fan of this, its clunky and strangely coupled

Reply:
`PersistEmbeddingsStateMachine(EmbeddingStateMachine)` inherits from `EmbeddingStateMachine` class and does not override these methods. We want this rule and all associated state to be exclusive to `EmbeddingStateMachine` and not `PersistEmbeddingsStateMachine`

Reply:
Can we use a mixin instead